source/elements/oneMKL/source/domains/spblas/operations/spmv.rst

```diff
@@ -0,0 +1,378 @@
+
+   * - Value
+     - Description
+     - Backend equivalent
+   * - ``default_alg``
+     - Default algorithm.
+     - | MKL: N/A
+       | cuSPARSE: ``CUSPARSE_SPMV_ALG_DEFAULT``
+       | rocSPARSE: ``rocsparse_spmv_alg_default``
+   * - ``coo_alg1``
+     - Default algorithm for COO format.
+     - | MKL: N/A
+       | cuSPARSE: ``CUSPARSE_SPMV_COO_ALG1``
+       | rocSPARSE: ``rocsparse_spmv_alg_coo``
+   * - ``coo_alg2``
+     - Deterministic algorithm for COO format.
+     - | MKL: N/A
+       | cuSPARSE: ``CUSPARSE_SPMV_COO_ALG2``
+       | rocSPARSE: ``rocsparse_spmv_alg_coo_atomic``
+   * - ``csr_alg1``
+     - Default algorithm for CSR format.
+     - | MKL: N/A
+       | cuSPARSE: ``CUSPARSE_SPMV_CSR_ALG1``
+       | rocSPARSE: ``rocsparse_spmv_alg_csr_adaptive``
+   * - ``csr_alg2``
+     - Deterministic algorithm for CSR format.
+     - | MKL: N/A
+       | cuSPARSE: ``CUSPARSE_SPMV_CSR_ALG2``
+       | rocSPARSE: ``rocsparse_spmv_alg_csr_stream``
+   * - ``csr_alg3``
+     - LRB variant of the algorithm for CSR format.
+     - | MKL: N/A
+       | cuSPARSE: N/A
+       | rocSPARSE: ``rocsparse_spmv_alg_csr_lrb``
+
+.. _onemkl_sparse_spmv:
+
+spmv
+----
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       void spmv_buffer_size(
+           sycl::queue                                &queue,
+           oneapi::mkl::transpose                     opA,
+           void*                                      alpha,
+           oneapi::mkl::sparse::matrix_view           A_view,
+           oneapi::mkl::sparse::matrix_handle_t       A_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+           void*                                      beta,
+           oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+           oneapi::mkl::sparse::spmv_alg              alg,
+           oneapi::mkl::sparse::spmv_descr_t          spmv_descr,
+           std::size_t                                &temp_buffer_size,
+       );
+
+       void spmv_optimize(
+           sycl::queue                                &queue,
+           oneapi::mkl::transpose                     opA,
+           void*                                      alpha,
+           oneapi::mkl::sparse::matrix_view           A_view,
+           oneapi::mkl::sparse::matrix_handle_t       A_handle,
+           oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+           void*                                      beta,
+           oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+           oneapi::mkl::sparse::spmv_alg              alg,
+           oneapi::mkl::sparse::spmv_descr_t          spmv_descr,
+           sycl::buffer<std::uint8_t, 1>              workspace,
+       );
+
+       sycl::event spmv_optimize(
+           sycl::queue                                &queue,
+           oneapi::mkl::transpose                     opA,
+           void*                                      alpha,
```

Comment:
None of the API contain const parameters to indicate pure input parameters, and I haven't seen (although there could be) a discussion about pointer aliasing (particularly in the case of USM).

Are these considerations to mention now? Pointer aliasing could be a section on its own, or can be discussed on a per-operation basis. Adding const to the API will help to show intended use of parameters.

I think it would be useful here, for example to show that `alpha` is a `const void *`. An implementation can do what it likes under the hood, but on return the value stored at memory address `*alpha` should be the same as on entry to `spmv_optimize`

Reply:
We discussed the idea of having const handle types but we decided this would not be needed for this version. I have only made the `alpha` and `beta` parameters const in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/32e43ffa1cf49ddf2050eb807c15043995ea1ef8